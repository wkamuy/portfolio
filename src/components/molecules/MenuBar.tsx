import { FC } from 'react';
import { Link as LinkRouter } from 'react-router-dom/';
import Link from '@material-ui/core/Link';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
// import CreateIcon from '@material-ui/icons/Create';
import GitHubIcon from '@material-ui/icons/GitHub';
import TwitterIcon from '@material-ui/icons/Twitter';
import PermIdentityIcon from '@material-ui/icons/PermIdentity';
import MenuBookIcon from '@material-ui/icons/MenuBook';
import Button from '@material-ui/core/Button';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import Drawer from '@material-ui/core/Drawer';
import Divider from '@material-ui/core/Divider';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    appBar: {
      backgroundColor: 'transparent',
      boxShadow: 'none',
      position: 'static',
    },
    iconButton: {
      marginRight: theme.spacing(2),
    },
    menuButtonBox: {
      marginLeft: '20px',
    },
    menuButton: {
      marginRight: theme.spacing(1),
      textTransform: 'none',
      color: 'inherit',
    },
    menuIcon: {
      color: 'white',
    },
    title: {
      fontFamily: ['Montserrat', 'sans-serif'].join(','),
    },
    snsIconBox: {
      marginLeft: 'auto',
    },
    list: {
      width: 170,
    },
    link: {
      textDecoration: 'none',
      color: 'white',
    },
  }),
);

type MenuBarProps = {
  title?: string;
  isOpen?: boolean;
  open?: (event: React.KeyboardEvent | React.MouseEvent) => void;
  close?: (event: React.KeyboardEvent | React.MouseEvent) => void;
};

const MenuBar: FC<MenuBarProps> = ({
  title,
  isOpen = false,
  open = () => undefined,
  close = () => undefined,
}) => {
  const classes = useStyles();

  // List of Side Bar
  const list = () => (
    <div
      className={classes.list}
      role="presentation"
      onClick={close}
      onKeyDown={close}
    >
      <List>
        <LinkRouter className={classes.link} to="/portfolio">
          <ListItem button>
            <ListItemIcon>
              <PermIdentityIcon />
            </ListItemIcon>
            <ListItemText primary="About" />
          </ListItem>
        </LinkRouter>
        <Divider variant="middle" />
        {/* <LinkRouter className={classes.link} to="/works">
          <ListItem button>
            <ListItemIcon>
              <CreateIcon />
            </ListItemIcon>
            <ListItemText primary="Works" />
          </ListItem>
        </LinkRouter> */}
        <Divider variant="middle" />
        <Link
          className={classes.link}
          href="https://wkamuy.hatenablog.com/"
          underline="none"
        >
          <ListItem button>
            <ListItemIcon>
              <MenuBookIcon />
            </ListItemIcon>
            <ListItemText primary="Blog" />
          </ListItem>
        </Link>
        <Divider variant="middle" />
      </List>
    </div>
  );

  return (
    <>
      <AppBar className={classes.appBar}>
        <Toolbar>
          <Box
            display={{
              xs: 'block',
              sm: 'block',
              md: 'none',
              lg: 'none',
              xl: 'none',
            }}
          >
            <IconButton
              className={classes.iconButton}
              aria-label="menu"
              onClick={open}
            >
              <MenuIcon className={classes.menuIcon} />
            </IconButton>
          </Box>
          <LinkRouter className={classes.link} to="/portfolio">
            <Typography variant="h6" className={classes.title}>
              {title}.PAGES
            </Typography>
          </LinkRouter>
          <Box
            display={{
              xs: 'none',
              sm: 'none',
              md: 'block',
              lg: 'block',
              xl: 'block',
            }}
            className={classes.menuButtonBox}
          >
            <Button className={classes.menuButton}>
              <LinkRouter className={classes.link} to="/portfolio">
                About
              </LinkRouter>
            </Button>
            {/* <Button className={classes.menuButton}>
              <LinkRouter className={classes.link} to="/works">
                Works
              </LinkRouter>
            </Button> */}
            <Button className={classes.menuButton}>
              <Link
                className={classes.link}
                href="https://wkamuy.hatenablog.com/"
                underline="none"
              >
                Blog
              </Link>
            </Button>
          </Box>
          <Box className={classes.snsIconBox}>
            <Link
              className={classes.link}
              href="https://twitter.com/wkamuy1"
              underline="none"
            >
              <IconButton>
                <TwitterIcon />
              </IconButton>
            </Link>
            <Link
              className={classes.link}
              href="https://github.com/wkamuy"
              underline="none"
            >
              <IconButton>
                <GitHubIcon />
              </IconButton>
            </Link>
          </Box>
        </Toolbar>
      </AppBar>
      <Drawer anchor="left" open={isOpen} onClose={close}>
        {list()}
      </Drawer>
    </>
  );
};

export default MenuBar;
