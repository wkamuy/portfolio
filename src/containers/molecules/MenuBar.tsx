import { FC } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { open, close } from 'actions';
import { MenuBarState } from 'reducer';
import MenuBar from 'components/molecules/MenuBar';

type EnhancedMenuBarProps = {
  title: string;
};

const EnhancedMenuBar: FC<EnhancedMenuBarProps> = ({ title }) => {
  const isOpen = useSelector<MenuBarState, boolean>((state) => state.isOpen);
  const dispatch = useDispatch();

  return (
    <MenuBar
      title={title}
      isOpen={isOpen}
      open={(event: React.KeyboardEvent | React.MouseEvent) =>
        dispatch(open(event))
      }
      close={(event: React.KeyboardEvent | React.MouseEvent) =>
        dispatch(close(event))
      }
    />
  );
};

export default EnhancedMenuBar;
