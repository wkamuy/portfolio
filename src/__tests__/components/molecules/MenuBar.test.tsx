import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import MenuBar from '../../../components/molecules/MenuBar';

describe('<MenuBar />', () => {
  it('sould render correctly', () => {
    const title = 'WKAMUY';
    const tree = renderer
      .create(
        <Router>
          <MenuBar title={title} />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
