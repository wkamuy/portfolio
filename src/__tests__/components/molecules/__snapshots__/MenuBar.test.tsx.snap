// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuBar /> sould render correctly 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-1 mui-fixed MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <div
      className="MuiBox-root MuiBox-root-10"
    >
      <button
        aria-label="menu"
        className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-2"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-menuIcon-5"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
      </button>
    </div>
    <h6
      className="MuiTypography-root makeStyles-title-6 MuiTypography-h6"
    >
      WKAMUY
      .PAGES
    </h6>
    <div
      className="MuiBox-root MuiBox-root-11 makeStyles-menuButtonBox-3"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-menuButton-4"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <a
            className="makeStyles-link-9"
            href="/portfolio"
            onClick={[Function]}
          >
            About
          </a>
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-menuButton-4"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineNone makeStyles-link-9 MuiTypography-colorPrimary"
            href="https://wkamuy.hatenablog.com/"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            Blog
          </a>
        </span>
      </button>
    </div>
    <div
      className="MuiBox-root MuiBox-root-12 makeStyles-snsIconBox-7"
    >
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineNone makeStyles-link-9 MuiTypography-colorPrimary"
        href="https://twitter.com/wkamuy1"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
              />
            </svg>
          </span>
        </button>
      </a>
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineNone makeStyles-link-9 MuiTypography-colorPrimary"
        href="https://github.com/wkamuy"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
              />
            </svg>
          </span>
        </button>
      </a>
    </div>
  </div>
</header>
`;
